enhancements/authentication/AuthConfig-missing-fields.md

```diff
@@ -0,0 +1,544 @@
+management cluster? How does it affect any components running split
+between the management cluster and guest cluster?
+
+#### Standalone Clusters
+
+Is the change relevant for standalone clusters?
+
+#### Single-node Deployments or MicroShift
+
+How does this proposal affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory?
+
+How does this proposal affect MicroShift? For example, if the proposal
+adds configuration options through API resources, should any of those
+behaviors also be exposed to MicroShift admins through the
+configuration file for MicroShift?
+
+### Implementation Details/Notes/Constraints
+
+What are some important details that didn't come across above in the
+**Proposal**? Go in to as much detail as necessary here. This might be
+a good place to talk about core concepts and how they relate. While it is useful
+to go into the details of the code changes required, it is not necessary to show
+how the code will be rewritten in the enhancement.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+### Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should
+_not_ be implemented.
+
+What trade-offs (technical/efficiency cost, user experience, flexibility,
+supportability, etc) must be made in order to implement this? What are the reasons
+we might not want to undertake this proposal, and how do we overcome them?
+
+Does this proposal implement a behavior that's new/unique/novel? Is it poorly
+aligned with existing user expectations?  Will it be a significant maintenance
+burden?  Is it likely to be superceded by something else in the near future?
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance,
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this?
+
+## Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+- What additional testing is necessary to support managed OpenShift service-based offerings?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+## Graduation Criteria
+
```

Comment:
This might require a bit more discussion, but should these new fields be added behind a feature-gate that users can toggle on and off beyond the existing `ExternalOIDC` feature gate? Why or why not?

My understanding is that new fields are typically added behind feature-gates so that we can gradually introduce them to users and ensure we have appropriate testing in place for the different behaviors.

It looks like the existing `ExternalOIDC` feature-gate is a default feature-gate in Hypershift, but still only in tech preview for standalone OpenShift. I think it might be OK to add these additional fields without an additional feature-gate but I'm still learning the ropes of feature-gate dos and don'ts. I'll reach out to Joel Speed to get some guidance.

Reply:
Reached out to Joel. It sounds like if we are waiting to promote the existing `ExternalOIDC` feature-gate to GA in standalone it should be OK to add these fields without an additional feature-gate. The `ExternalOIDC` feature just wouldn't be able to go GA until these new fields have been added and functionality is implemented and tested.

If we promote the `ExternalOIDC` feature-gate to GA before adding these fields, we will need to add a new feature-gate with a dependence on the existing `ExternalOIDC` feature-gate.